src/rgw/rgw_sal_motr.cc

```diff
@@ -1294,11 +1324,6 @@ int MotrObject::get_obj_attrs(RGWObjectCtx* rctx, optional_yield y, const DoutPr
 int MotrObject::modify_obj_attrs(RGWObjectCtx* rctx, const char* attr_name, bufferlist& attr_val, optional_yield y, const DoutPrefixProvider* dpp)
 {
   rgw_obj target = get_obj();
-  int r = get_obj_attrs(rctx, y, dpp, &target);
-  if (r < 0) {
-    return r;
```

Comment:
instead of removing, just log that get_obj_attrs() failed, so we are dealing with new attribute set possibly, continuing with set attributes.

Reply:
Not removing this block of code. I was initially experiencing some strange behavior in log statement. Removing this resolved me issues in log. 
Now After removing if **(this->store->get_obj_meta_cache()->get(dpp, key, bl))**  in set_obj_attrs call, issues in log is not seen
Reverted back to original code

Reply:
sure, thats fine then